File: iosgl/iosgl/bt/display/role/vo/MeshData.m
//
//  MeshData.m
//  iosgl
//
//  Created by zhao on 2/4/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "MeshData.h"
#import "MaterialBaseParam.h"
#import "Material.h"

@interface MeshData()

@property(nonatomic,strong)NSMutableArray<NSNumber*>* boneIDAry;
@property(nonatomic,strong)NSMutableArray<NSNumber*>* boneWeightAry;
@property(nonatomic,strong)NSMutableArray<NSNumber*>* boneNewIDAry;

@property(nonatomic,strong)NSMutableArray<NSNumber*>* materialParamData;
@property(nonatomic,strong)MaterialBaseParam* materialParam;
@property(nonatomic,strong)Material* material;
@property(nonatomic,strong)NSString* materialUrl;

@property(nonatomic,assign)GLuint boneWeightBuffer ;
@property(nonatomic,assign)GLuint boneIdBuffer ;
 
@property(nonatomic,assign)float  uid;
@property(nonatomic,assign)float  boneIDOffsets;
@property(nonatomic,assign)float  boneWeightOffsets;
 
@end

@implementation MeshData

@end

File: testproject/testdemo/app/src/androidTest/java/com/example/pan3d/base/MeshData.java
package com.pan3d.base;

import com.pan3d.material.Material;
import com.pan3d.material.MaterialBaseParam;
import com.pan3d.res.MaterialInfoVo;
import com.pan3d.scene.Scene3D;
import com.pan3d.vo.BindParticle;
import com.pan3d.vo.Matrix3D;

import java.nio.FloatBuffer;
import java.util.ArrayList;
import java.util.List;

public class MeshData extends ObjData {
    public  List<Float>  boneIDAry ;
    public  List<Float>  boneWeightAry;
    public FloatBuffer boneWeightBuffer ;
    public FloatBuffer boneIdBuffer ;
    public  List<Short>  boneNewIDAry;
    public String materialUrl;
    public List<MaterialInfoVo> materialParamData  ;
    public MaterialBaseParam materialParam;
    public List<BindParticle> particleAry;
    public int uid;
    public int boneIDOffsets;
    public int boneWeightOffsets;
    public Material material;

    public List<Matrix3D> bindPosAry ;

    public MeshData(Scene3D val){
        super(val);
        this.particleAry=new ArrayList<>();
    }
    public void  upToGup()
    {
       if( this.isCompile){
           return;
       }
        this.isCompile=true;
        if(this.verticeslist!=null){
            this.vertexBuffer= ObjData.upGpuvertexBuffer(this.verticeslist);
        }
        if(this.uvlist!=null){
            this.uvBuffer= ObjData.upGpuvertexBuffer(this.uvlist);
        }
        if(this.boneIDAry!=null){
            this.boneIdBuffer= ObjData.upGpuvertexBuffer(this.boneIDAry);
        }
        if(this.boneWeightAry!=null){
            this.boneWeightBuffer= ObjData.upGpuvertexBuffer(this.boneWeightAry);
        }
        if(this.indexs!=null){
            this.indexBuffer= ObjData.upGpuIndexBuffer(this.indexs);
            this.treNum= this.indexs.size();
        }


    }

}
